shell/common/shell.cc

```diff
@@ -610,6 +611,18 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  // Only set this if the user has _not_ requested a specific value.
+  if (raster_cache_max_bytes_user_value_ == -1) {
+    // This is the formula Android uses.
+    int max_bytes = metrics.physical_width * metrics.physical_height * 12 * 4;
```

Comment:
Add shell unit tests for this? You can go over the 4K, 5K, and new mac display settings there :)

Reply:
Will stop by - I'm struggling a bit on the test for this since things fizzle if there's no GrContext.